web/app/components/tools/edit-custom-collection-modal/index.tsx

```diff
@@ -87,16 +87,17 @@ const EditCustomCollectionModal: FC<Props> = ({
       return
     }
     (async () => {
-      const customCollection = getCustomCollection()
       try {
         const { parameters_schema, schema_type } = await parseParamsSchema(debouncedSchema)
+        const customCollection = getCustomCollection()
```

Comment:
I don't think it should be changed like this, and the code before the try catch should have the same effect as the current code. Maybe you should update the deps instead.

```
React Hook useEffect has missing dependencies: 'editFirst', 'getCustomCollection', 'isEdit', and 'setCustomCollection'. Either include them or remove the dependency array.
```


```
  useEffect(() => {
    if (!debouncedSchema)
      return
    if (isEdit && editFirst) {
      setEditFirst(false)
      return
    }
    (async () => {
      const customCollection = getCustomCollection()
      try {
        const { parameters_schema, schema_type } = await parseParamsSchema(debouncedSchema)
        const newCollection = produce(customCollection, (draft) => {
          draft.schema_type = schema_type
        })
        setCustomCollection(newCollection)
        setParamsSchemas(parameters_schema)
      }
      catch (e) {
        const newCollection = produce(customCollection, (draft) => {
          draft.schema_type = ''
        })
        setCustomCollection(newCollection)
        setParamsSchemas([])
      }
    })()
  }, [debouncedSchema, editFirst, getCustomCollection, isEdit, setCustomCollection])
```

Reply:
> the code before the try catch should have the same effect as the current code.

no, issue accur because `getCustomCollection` was called before await. assume this behavior in browser:
1. old = getCustomCollection() // {schema: "abc"}
2. user input change the value state // value is {schema: "abcd"} whatever
3. setCustomCollection(old) // value is old, user input change will be lost
4. Because the code set the old value to `schema`, in the meantime when I enter input, just cause the problem with #6395.

> Maybe you should update the deps instead.

This useEffect seems that logic is watching the `debouncedSchema`, if it changes, call api and parse params schema, it is totally a side effect, I think it is not a good idea to set other deps to trigger api call, i think the logic is clear?

Reply:
👍